packages/bot-web-ui/src/components/dashboard/dashboard.tsx

```diff
@@ -96,53 +99,76 @@ const Dashboard = ({
         }
     };
 
-    let getBotBuilderToken: string | number = '';
-    let getOnboardingToken: string | number = '';
+    const is_tour_complete = React.useRef(true);
+
+    let bot_tour_token: string | number = '';
+    let onboard_tour_token: string | number = '';
 
     React.useEffect(() => {
         if (active_tab === 0 && has_file_loaded) {
             onEntered();
         }
         if (active_tab === 0) {
-            getOnboardingToken = getTourSettings('token');
-            setOnBoardingTokenCheck(getOnboardingToken);
+            setTourType('onboard_tour_');
+            onboard_tour_token = getTourSettings('token');
+            setOnBoardingTokenCheck(onboard_tour_token);
         }
         if (active_tab === 1 && !has_onboard_tour_started) {
             setTourType('bot_builder_');
-            getBotBuilderToken = getTourSettings('token');
-            setBotBuilderTokenCheck(getBotBuilderToken);
-            setTourDialogVisibility(true);
+            bot_tour_token = getTourSettings('token');
+            setBotBuilderTokenCheck(bot_tour_token);
         }
-
         tour_status = getTourSettings('onboard_tour_status');
         setTourStatus(tour_status);
-    }, [active_tab, handleJoyrideCallback, has_onboard_tour_started]);
+    }, [active_tab, handleJoyrideCallback, has_onboard_tour_started, is_tour_complete, tour_status_ended]);
```

Comment:
Please remove the useRef variable from the useEffect Dependency list.
Its not a good practice, Found this article if you wanna read more: https://blog.bitsrc.io/do-you-make-this-mistake-with-useeffect-in-react-5679eeb8304a

@farabi-deriv